src/pyhf/optimize/opt_minuit.py

```diff
@@ -102,8 +105,11 @@ def _minimize(
         # 0: Fast, user-provided gradient
         # 1: Default, no user-provided gradient
         strategy = options.pop(
-            'strategy', self.strategy if self.strategy is not None else not do_grad
+            'strategy', self.strategy if self.strategy is not None else int(not do_grad)
         )
+        # Passing strategy=None to options requires another check to guard against None
+        if strategy is None:
+            strategy = int(not do_grad)
```

Comment:
This is done _after_ the above and not _before_ by using

```python
        if self.strategy is None:
            self.strategy = int(not do_grad)
```

to avoid mutating `self.strategy` making it harder to trace during debugging.